### PR TITLE
[Database] Refactored Database Architecture

### LINKED ISSUE #172: Revamp Database Architecture
**Is your feature request related to a problem? Please describe.**
Currently the structure or architecture of database contains a single User class for every kind of roles i.e Admin, Extractor and Management.

**Describe the solution you'd like**
This structure could be improved by following trivial approach creating designing Database architecture. 

**Describe alternatives you've considered**
class_diagram

**Additional context**

### PR DESCRIPTION
**Pull request by sachinsom93**
# Description
This PR contains the following changes/commits:

**Commit 1**
Implemented Class definition
        * BaseUser
        * AdminUser
        * ExtractorUser
        * ManagementUser

**Commit 2**
 Implemented Class Definitions of the following:
    * Base_user (abstract Class)
    * Admin Model
    * Extractor Model
    * Management Model
    * Case Model
    * Task Model

**Commit 3**
    Implemented schemas for serialization of:
    * Admin,
    * Extractor,
    * Management,
    * Case,
    * Task

**Commit 4**
 Modified the existing routes according
    to new database architecture:
    * Route for user management,
    * Route for extraction,
    * Route for cases

**Commit 5**
 Updated requirements.txt files

**Commit 6**
 Modified authentication routes according new database architecture.

**Commit 7**
 Updated database migrations.

**Commit 8**
Updated endpoints.txt file flask-backend.

**Commit 9**
Corrected minor spell error

**Commit 10**
Updated flask-backend/Readme for flask backend setup.

Fixes #172 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?
Using Postman.

# Screenshots
**Admin Registration route**
image

**User Login route**
image

**User profile route**
image

**Add member route that would be used by admin to add new members**
image

image

**Get route for listing all members of an admin**
image

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by sachinsom93** [comment]
Could you please review this, sir @shivanshu1333 ?

File: flask-backend/requirements.txt [hunk 1]
```diff
@@ -1,20 +1,31 @@
+alembic==1.6.5
```
Comment (shivanshuraj1333) [inline flask-backend/requirements.txt]: Remove redundant and duplicate imports

File: flask-backend/migrations/versions/989dbc01a9b0_.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+"""empty message
```
Comment (shivanshuraj1333) [inline flask-backend/migrations/versions/989dbc01a9b0_.py:1]: Do we need to push this?

Comment (sachinsom93) [reply flask-backend/migrations/versions/989dbc01a9b0_.py:1]: Yes, it's a part of migrations file, that contains versions of database migrations.

File: flask-backend/requirements.txt (hunk 1 above)
Comment (sachinsom93) [reply flask-backend/requirements.txt]: @shivanshu1333 sir, these were installed automatically when I installed Flask-Migrate, python-dotenv packages.

Comment (sachinsom93) [reply flask-backend/requirements.txt]: And some of them were of #173, that I forgot to add previously :)

Comment (shivanshuraj1333) [reply flask-backend/requirements.txt]: there are multiple duplicate imports like for Flask-Login==0.5.0,
okay you can do one thing, locally delete the requirements.txt file and inside the venv of flask do pip freeze > requirements.txt and push the file

Comment (sachinsom93) [reply flask-backend/requirements.txt]: sure sir @shivanshu1333, I have deleted requirements.txt and created requirements.txt where venv is located.

Comment (sachinsom93) [reply flask-backend/requirements.txt]: Done @shivanshu1333 !, I have removed requirements.txt file from flask-backend, now there is only one requirements.txt file in the same level of venv :)

**Review by shivanshuraj1333 — APPROVED** [review]

**Pull request merged**